src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -1932,7 +1937,7 @@ private void ReadNextNestedResourceInfoContentItemInRequest()
             if (nestedResourceInfo.HasEntityReferenceLink)
             {
                 this.EnterScope(new Scope(ODataReaderState.EntityReferenceLink,
-                    nestedResourceInfo.ReportEntityReferenceLink(), null, null, this.CurrentScope.ODataUri));
+                    nestedResourceInfo.ReportEntityReferenceLink(), null, (IEdmStructuredTypeReference) null, this.CurrentScope.ODataUri));

```

Comment:
>(IEdmStructuredTypeReference) [](start = 74, length = 29)

I don't like it. Do we have other solutions?

Reply:
Added another constructor.

---
In reply to: [263957299](https://github.com/OData/odata.net/pull/1322#discussion_r263957299) [](ancestors = 263957299)